### DISCUSSION TITLE
Delete item function with useSWRmutation hook

### DISCUSSION THREAD

**Discussion by Olga-Mishareva** [neuefische/questions#516] [Web] [answered]
Hi, 

I made removing an item using `useSWRmutation` hook and at first everything worked fine, but when I started adding `useStore` with logic to confirm deleting (now commented out), then the removal of the item itself went fine, but then the request for the already removed id happens again and the application crashes with 
 `Uncaught Error: Objects are not valid as a React child (found: [object Error]). If you meant to render a collection of children, use an array instead.`

I saved and [pushed actual State](https://github.com/Olga-Mishareva/capstone-project-lost-n-found/tree/refactoring) in the branch `refactoring`.

I can see from the console.log that a repeated request is happening, but I don't understand how I can fix it.

Thank you for your help!

**Comment by F-Kirchhoff** [comment, marked as answer]
The problem is, that you have to add an option when you call the trigger function of your `useSWRMutation` hook. Because you delete the item the default behaviour of useSWR of revalidating the data of the item must be turned of. This revalidating is causing a get request for the just deleted item which resloved in your unsuccessful fetch request you have seen. By turning the revalidation off when you trigger a `delete` mutation, useSWR is not performing this `get` request.
```js
const {trigger} = useSWRMutation(`api/items/${id}`,fetcher)

...

async function handleDelete() {
  await trigger({method: 'DELETE'},{revalidate: false})
}
```
The other problem was that you tried to render the error from the `useSWRMutation` hook directly in an `h2` tag. Because this error is an object and objects are not allowed as children you either need to convert this object to a string or just render the `error.message`.
